tasks.py

```diff
@@ -1643,3 +1771,182 @@ def get_command_version(get_version_command: str, version_prefix: str) -> Option
+        hide=True,
+        warn=True,
+    )
+
+    if bgp_type_result.ok:
+        value = bgp_type_result.stdout.strip()
+        # 'native' if not set (follows the Go code behaviour)
+        return value if value else "native"
+    else:
+        raise Exception(
+            f"Cannot detect BGP type from controller deployment in {namespace} namespace: command failed"
+        )
+
+
+def detect_prometheus():
+    """Detect if Prometheus is available by checking monitoring namespace."""
+    prom_result = run(f"{kubectl_path} get namespace monitoring", hide=True, warn=True)
+    return prom_result.ok
+
+
+def detect_ip_family(namespace="metallb-system"):
+    """Detect IP family from ipaddresspool resource (dual, ipv4, ipv6)."""
+    addresses_result = run(
+        f"{kubectl_path} get ipaddresspool -n {namespace} "
+        f"-o jsonpath='{{.items[*].spec.addresses[*]}}'",
+        hide=True,
+        warn=True,
+    )
+
+    if not addresses_result.ok:
+        raise Exception(
+            f"Cannot detect IP family from ipaddresspool resource in {namespace} namespace: command failed"
+        )
+
+    addresses = addresses_result.stdout.strip().split()
+    if not addresses:
+        raise Exception(
+            "Cannot detect IP family from ipaddresspool resource: no addresses found"
+        )
+
+    has_ipv4 = False
+    has_ipv6 = False
+
+    for address in addresses:
+        try:
+            # Handle IP ranges (e.g., "172.18.0.9-172.18.0.19")
+            if "-" in address:
+                start_ip, end_ip = address.split("-", 1)
+                start = ipaddress.ip_address(start_ip.strip())
+                end = ipaddress.ip_address(end_ip.strip())
+
+                # Both IPs must be same version and start <= end
+                if start.version != end.version:
+                    raise Exception(f"IP range '{address}' contains mixed IP versions")
+                if start > end:
+                    raise Exception(
+                        f"IP range '{address}' has start IP greater than end IP"
+                    )
+
+                # Use the start ip as a representative address for version checking
+                address = start
+
+            # Handle single IPs or CIDR networks
+            if is_ipv4(address):
+                has_ipv4 = True
+            elif is_ipv6(address):
+                has_ipv6 = True
+
+        except ValueError as e:
+            raise Exception(
+                f"Invalid IP address or network '{address}' found in ipaddresspool: {e}"
+            )
+
+    if has_ipv4 and has_ipv6:
+        return "dual"
+    elif has_ipv6:
+        return "ipv6"
+    elif has_ipv4:
+        return "ipv4"
+    else:
+        raise Exception(
+            "Cannot detect IP family from ipaddresspool resource: unmatched ip family addresses"
+        )
+
+
+def detect_protocol(namespace="metallb-system"):
```

Comment:
for the same reasons, but this time it is not really detectable 😅 I think we should drop this and let the user decide